### PR TITLE
[275] Update documentation of the occ config:list command

### LINKED ISSUE #275: Doc improvements for commands occ config:xxx necessary
The command set config:xxx is documented vary basic and not with all parameters available.
Pls see:  https://github.com/owncloud/documentation/blob/master/admin_manual/configuration/server/occ_command.rst#L329

I came across this with https://github.com/owncloud/core/issues/32006

**Based on the example:**
```
sudo -uwww-data ./occ config:list --help

Usage:
  config:list [options] [--] [<app>]

Arguments:
  app                    Name of the app ("system" to get the config.php values, "all" for all apps and system). [default: "all"]

Options:
      --output[=OUTPUT]  The output format to use (plain, json or json_pretty, default is plain). [default: "json_pretty"]
      --private
```

**But this is currently documented:**
```
sudo -u www-data php occ config:list

By default, passwords and other sensitive data are omitted from the report, so the output can be posted publicly (e.g., as part of a bug report). 
In order to generate a full backport of all configuration values the ``--private`` flag needs to be set:
```

**Missing in the example above:**
- ``app``
- ``--output[=OUTPUT]``

In addition, if I take the example from the referenced core issue:
```
./occ config:list files_sharing
{
    "apps": {
        "files_sharing": {
            "enabled": "yes",
            "incoming_server2server_share_enabled": "no",
            "installed_version": "0.10.1",
            "outgoing_server2server_share_enabled": "no",
            "types": "filesystem"
        }
    }
}
```
How to identify the app key where you can apply changes?
Which datatypes and/or values are allowed for a key?
I do not believe that eg ``installed_version`` is a changeable key,
but ``outgoing_server2server_share_enabled`` is...
Pls see also the new key ``shareapi_public_notification_lang`` at https://github.com/owncloud/core/pull/31869

(Maybe config:list needs and extension with a new option to show which datatypes and/or values are allowed for a key)

@phil-davis @PVince81

### PR DESCRIPTION
**Pull request by settermjd**
This change:

- documents the ability to filter the command's output
- shows a complete report example (with sensitive information removed); and
- shows how to include sensitive information in the generated output/report

This fixes #275.

### CODE REVIEW

File: modules/admin_manual/pages/configuration/server/occ_command.adoc [hunk 1]
```diff
@@ -376,21 +376,67 @@ config
...
+You can see an sample output in the example below.
+
+[source,json]
+----
+include::{partialsdir}configuration/server/occ_command/config-list-report.json[]
+----
+
+===== Displaying Sensitive Information
+
+In order to generate a full backport which includes sensitive values, such as passwords and salts, use the `--private` option, as in the follow example.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/server/occ_command.adoc]:
```suggestion
In order to generate a full report which includes sensitive values, such as passwords and salts, use the `--private` option, as in the follow example.
```

**Comment by mmattel** [comment]
The descriptiojn is incomplete.
The command help output is much more:
```
./occ config:list --help
Usage:
  config:list [options] [--] [<app>]

Arguments:
  app                    Name of the app ("system" to get the config.php values, "all" for all apps and system). [default: "all"]

Options:
      --output[=OUTPUT]  The output format to use (plain, json or json_pretty). [default: "json_pretty"]
      --private          Use this option when you want to include sensitive configs like passwords and salts.
```
You need to add Arguments/Options like just above this command (background:queue:execute). Just c/p and adopt.

Sidenote: just seeing that `config:import`is also missing the Arguments table. You could fix that in one shot. (yes I know could be another PR too but...).
```
./occ config:import --help
Usage:
  config:import [<file>]

Arguments:
  file                  File with the JSON array to import.
```

**Comment by settermjd** [comment]
> The descriptiojn is incomplete.
> The command help output is much more:
> 
> ```
> ./occ config:list --help
> Usage:
>   config:list [options] [--] [<app>]
> 
> Arguments:
>   app                    Name of the app ("system" to get the config.php values, "all" for all apps and system). [default: "all"]
> 
> Options:
>       --output[=OUTPUT]  The output format to use (plain, json or json_pretty). [default: "json_pretty"]
>       --private          Use this option when you want to include sensitive configs like passwords and salts.
> ```
> 
> You need to add the Arguments/Options table like just above at command (background:queue:execute). Just c/p and adopt.
> 
> Sidenote: just seeing that `config:import`is also missing the Arguments table. You could fix that in one shot. (yes I know could be another PR too but...).
> 
> ```
> ./occ config:import --help
> Usage:
>   config:import [<file>]
> 
> Arguments:
>   file                  File with the JSON array to import.
> ```

You're right, it would be another PR :grin:. That said, I'll get the table included in the PR.

**Comment by settermjd** [comment]
@mmattel, I've added the arguments and options tables. In addition to that, I've combined it into the previous commits, which I combined together, to ensure that the history is as clean and organised as possible.

File: modules/admin_manual/pages/configuration/server/occ_command.adoc [hunk 2]
```diff
@@ -376,21 +376,83 @@ config
...
 ----
 
-You can list all configuration values with one command:
+==== config:list
+
+The `config:list` command lists all configuration values, both for your ownCloud setup, along with any apps.
 
 [source,console,subs="attributes+"]
 ----
 {occ-command-example-prefix} config:list
```
Comment (mmattel) [inline modules/admin_manual/pages/configuration/server/occ_command.adoc]: `{occ-command-example-prefix} config:list [options] [--] [<app>] `

File: modules/admin_manual/pages/configuration/server/occ_command.adoc [hunk 3]
```diff
@@ -376,21 +376,83 @@ config
...
+
+[width="100%",cols="20%,70%",]
+|===
+| `--private`
+| Use this option when you want to include sensitive configs, like passwords and salts.
+|===
+
 By default, passwords and other sensitive data are omitted from the report, so the output can be posted publicly (e.g., as part of a bug report). 
-In order to generate a full backport of all configuration values the `--private` flag needs to be set:
+You can see an sample output in the example below.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/server/occ_command.adoc]:
```suggestion
You can see sample output in the example below.
```

File: modules/admin_manual/pages/configuration/server/occ_command.adoc [hunk 4]
```diff
@@ -376,21 +376,83 @@ config
...
+
+[source,json]
+----
+include::{partialsdir}configuration/server/occ_command/config-list-report.json[]
+----
+
+===== Displaying Sensitive Information
+
+In order to generate a full report which includes sensitive values, such as passwords and salts, use the `--private` option, as in the follow example.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/server/occ_command.adoc]:
```suggestion
In order to generate a full report which includes sensitive values, such as passwords and salts, use the `--private` option, as in the following example.
```

File: modules/admin_manual/pages/configuration/server/occ_command.adoc (hunk 2 above)
Comment (settermjd) [reply modules/admin_manual/pages/configuration/server/occ_command.adoc]: Thanks for pointing that out.

**Pull request merged**